app/javascript/components/legacy/nonprofits/donate/InfoStep.tsx

```diff
@@ -0,0 +1,379 @@
+// License: LGPL-3.0-or-later
+import { Money } from '../../../../common/money';
+import React, { useContext, useState } from 'react';
+import { SupporterType, RequiredFieldsType, ActionType, AddressProps } from './wizard';
+import { WizardContext } from '../../_dependencies/ff-core/wizard';
+import { Formik, useFormikContext } from 'formik';
+import { useIntl } from "../../../intl";
+
+// const h = require('snabbdom/h')
+// const R = require('ramda')
+// const flyd = require('flyd')
+// const uuid = require('uuid')
+// const supporterFields = require('../../components/supporter-fields')
+// const button = require('ff-core/button')
+// const dedicationForm = require('./dedication-form')
+// const serialize = require('form-serialize')
+// const request = require('../../common/request')
+
+// const sepaTab = 'sepa'
+// const cardTab = 'credit_card'
+
+interface InfoStepProps {
+  loading: boolean;
+  error: string;
+  loadingText: string;
+  address: AddressProps;
+  required: RequiredFieldsType;
+  supporter: SupporterType;
+  hideDedication: boolean;
+  isRecurring: boolean;
+  weekly: boolean;
+  amount: Money;
+  currencySymbol: string;
+  stateDispatch: (action: ActionType) => void;
+}
+
+interface FormikFormValues {
+  selectedPayment: string;
+  supporter: SupporterType;
+  address: AddressProps;
+}
+
+export function InfoStep(props: InfoStepProps): JSX.Element {
+	const stepManagerContext = useContext(WizardContext);
+
+	return (
+		<div className={'wizard-step info-step u-padding--10'}>
+			<Formik onSubmit={(values) => {
+				// post supporter data
+				props.stateDispatch({
+					type: 'setSelectedPayment',
+					selectedPayment: values.selectedPayment,
+				});
+				props.stateDispatch({
+					type: 'setSupporter',
+					supporter: values.supporter,
+					address: values.address,
+					next: stepManagerContext.next,
+				});
+			}} initialValues={{ supporter: props.supporter, address: props.address } as FormikFormValues}>
```

Comment:
Important note: in React you should always set the form field default values; if the value is null or undefined, then it should be a blank string. React has problems tracking form field values when they're originally set to null or undefined (I don't totally understand it but every bit of advice has told me that).

If the service or whatever is getting the result of the submit expects that those values will be null or undefined, then you should convert empty form fields back to null or undefined on submit.